src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -0,0 +1,56 @@
+package com.tstepnik.planner.service;
+
+import com.tstepnik.planner.domain.Importance;
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.exceptions.UserNotFoundException;
+import com.tstepnik.planner.repository.TaskRepository;
+import com.tstepnik.planner.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private final Importance DEFAULT_IMPORTANCE = Importance.INCIDENTAL;
+
+    private final TaskRepository taskRepository;
+    private final UserRepository userRepository;
+
+    public TaskService(TaskRepository taskRepository, UserRepository userRepository) {
+        this.taskRepository = taskRepository;
+        this.userRepository = userRepository;
+    }
+
+    public List<Task> getTasks() {
+        return taskRepository.findAll();
+    }
+
+    public Task addTask(Task task, Principal principal) {
+        task.setImportance(DEFAULT_IMPORTANCE);
+        task.setUser(userRepository.findByLogin(principal.getName()).get());
+        return taskRepository.save(task);
+    }
+
+    public Task updateTask(Task task, Long id, Principal principal) {
+        Optional<Task> userTask = taskRepository.findUserTask(id, principal.getName());
```

Comment:
Good idea but I would do it differently. find task in db by id, check if this tasks user is the same as principal, if so, continue to update if not throw exception.

Reply:
Is correct what I did ?

Reply:
I add commit, just check it.

Reply:
The changes are in last commit.